packages/expo-firebase-analytics/src/ExpoFirebaseAnalytics.web.ts

```diff
@@ -0,0 +1,65 @@
+// import * as firebase from "firebase/app";
+// // side-effect
+// import "firebase/analytics";
+
+function getFirebaseModule() {
+  try {
+    return require('firebase/app');
+  } catch ({ message }) {
+    throw new Error('Firebase JS SDK is not installed: ' + message);
+  }
+}
+function getAnalyticsModule() {
+  try {
+    const firebase = getFirebaseModule();
+    require('firebase/analytics');
+    return firebase.analytics();
+  } catch ({ message }) {
+    throw new Error('Firebase JS Analytics SDK is not available: ' + message);
+  }
+}
+
+export default {
+  get name(): string {
+    return 'ExpoFirebaseAnalytics';
+  },
+  async initializeAppDangerously(config: { [key: string]: any }): Promise<void> {
+    getFirebaseModule().initializeApp(config);
```

Comment:
This function (and others) don't return `Promise<void>`, but `void` and the promise isn't awaited. Fix as you wish (`await` or `return`).

Reply:
This is something we do as a standard in all of our modules to ensure they return a Promise.

Reply:
Sorry, I was too used to methods to be asynchronous that it didn't even cross my mind that `firebase.app.initializeApp()` could be synchronous. Resolving conversation